Sources/ScheduledTasks.php

```diff
@@ -1246,12 +1246,16 @@ function scheduled_fetchSMfiles()
 	foreach ($js_files as $ID_FILE => $file)
 	{
 		// Create the url
-		$server = empty($file['path']) || (substr($file['path'], 0, 7) != 'http://' && substr($file['path'], 0, 8) != 'https://') ? 'http://www.simplemachines.org' : '';
+		$server = empty($file['path']) || (substr($file['path'], 0, 7) != 'http://' && substr($file['path'], 0, 8) != 'https://') ? 'https://www.simplemachines.org' : '';
 		$url = $server . (!empty($file['path']) ? $file['path'] : $file['path']) . $file['filename'] . (!empty($file['parameters']) ? '?' . $file['parameters'] : '');
 
 		// Get the file
 		$file_data = fetch_web_data($url);
 
+		// If we got an error - maybe https isn't working, try http?
+		if ($file_data === false && substr($url, 0, 30) == 'https://www.simplemachines.org')
+			$file_data = fetch_web_data(str_replace('https://www.simplemachines.org', 'http://www.simplemachines.org', $url));
```

Comment:
If you got an idea of how to better validate the file, we can make things change for 2.1.  Previous versions of SMF still check against http.

I'm wanting to convert the latest-*.js files to be json based rather than js based.  But they currently do a little more than what you can do with json based results including handling some unique javascript coding to make things pop better in the admin panel for update warnings.